client/interactive.js

```diff
@@ -9,42 +9,40 @@ export class FormulaBunBotInteracive extends FormulaBunBotBase {
     super(param);
     this.players = {};
     this.server = {};
-    this.on("ready", () => {
-      this.registercommands();
-      this.ws.on("INTERACTION_CREATE", (interaction) =>
-        this.respond(interaction)
-      );
+    this.on('ready', async () => {
+      await this.registercommands();
+      this.on('interactionCreate', (interaction) => {
+        if (interaction.isCommand()) {
+          this.respond(interaction);
+        }
+      });
     });
   }
 
-  registercommands() {
+  async registercommands() {
     for (let c in commands) {
       if (commands.hasOwnProperty(c)) {
-        let guild,
-          application = this.api.applications(this.user.id);
-
+        let commandsObject;
         switch (process.env.DISCORDBOT_ENV) {
-          case "test":
-            guild = application.guilds(TEST_GUILD);
+          case 'test':
+            commandsObject = this.application.commands;
             break;
-          case "deploy":
-            guild = application;
+          case 'deploy':
+            commandsObject = (await this.guilds.fetch(TEST_GUILD)).commands;
             break;
           default:
-            throw "DISCORDBOT_ENV not set, possible values: test, deploy.";
+            throw 'DISCORDBOT_ENV not set, possible values: test, deploy.';
         }
 
-        guild.commands
-          .post({
-            data: {
-              name: c,
-              description: commands[c].descr,
-            },
-          })
-          .then(() => {
-            console.log(`registered ${c}`);
-          })
-          .catch(console.error);
+        try {
+          await commandsObject.create({
+            name: c,
+            description: commands[c].descr,
+          });
+          console.log(`resistered ${c}`);
+        } catch (err) {
+          console.error(err);
+        }
```

Comment:
what are the benefits of `try catch` over `.catch`. `commandsObject.create` is a promise  catch is more natural to me.

Reply:
I honestly didn't have much of a reason and just assumed that was the right thing to do. Been trying to avoid using promise structure but if you think that it would be clearer then I don't mind using it.